SofaKernel/modules/SofaBaseMechanics/BarycentricMappers/BarycentricMapperTetrahedronSetTopology.h

```diff
@@ -68,6 +68,107 @@ class BarycentricMapperTetrahedronSetTopology : public BarycentricMapperTopology
     void computeDistance(double& d, const Vector3& v) override;
     void addPointInElement(const int elementIndex, const SReal* baryCoords) override;
 
+    //handle topology changes depending on the topology
+    void processTopologicalChanges(const typename Out::VecCoord& out, const typename In::VecCoord& in, core::topology::Topology* t) {
+        using sofa::core::behavior::MechanicalState;
+
+        if (t != this->m_toTopology) return;
+
+        if ( this->m_toTopology->beginChange() == this->m_toTopology->endChange() )
+            return;
+
+        std::list<const core::topology::TopologyChange *>::const_iterator itBegin = this->m_toTopology->beginChange();
+        std::list<const core::topology::TopologyChange *>::const_iterator itEnd = this->m_toTopology->endChange();
+
+        helper::vector<MappingData>& vectorData = *(d_map.beginEdit());
```

Comment:
Use WriteAccessor instead of beginEdit()/endEdit() this is shorter to write and more safer with respect to security.